pkg/apis/core/v1/defaults.go

```diff
@@ -162,6 +162,11 @@ func SetDefaults_Pod(obj *v1.Pod) {
 		enableServiceLinks := v1.DefaultEnableServiceLinks
 		obj.Spec.EnableServiceLinks = &enableServiceLinks
 	}
+	if obj.Spec.Preempting == nil {
```

Comment:
put a feature gate check around this defaulting and only populate the field if the `NonPreemptingPriority` feature is enabled

(same comments apply to PriorityClass defaults)

Reply:
that means the places where we check this field need to deal with an empty value coming from the apiserver, at least until the feature is enabled by default (and possible one release past that, to deal with HA scheduler/apiserver skew)

Reply:
Done